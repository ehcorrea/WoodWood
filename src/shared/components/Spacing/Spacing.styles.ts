import styled, { css } from '@emotion/native';

type ContainerProps = {
  x: number;
  y: number;
};

export const Container = styled.View<ContainerProps>`
  ${({ x, y }) => css`
    margin-horizontal: ${`${x}px`};
    margin-vertical: ${`${y}px`};
  `}
`;
