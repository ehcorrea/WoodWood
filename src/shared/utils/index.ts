export * from './carrinho/carrinho';
export * from './responsive/responsive';
