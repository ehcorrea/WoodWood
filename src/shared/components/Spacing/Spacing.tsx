import * as S from './Spacing.styles';

export type SpacingProps = {
  x?: number;
  y?: number;
};

export function Spacing({ x = 0, y = 0 }: SpacingProps) {
  return <S.Container testID="spacing" x={x} y={y} />;
}
