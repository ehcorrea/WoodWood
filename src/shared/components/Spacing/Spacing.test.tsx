import { render } from '@/test/utils';

import { Spacing, SpacingProps } from './Spacing';

const setup = (props?: SpacingProps) => render(<Spacing {...props} />);

describe('<Spacing/>', () => {
  describe('quando renderizado', () => {
    test.each([
      { props: { x: 5, y: 5 }, result: { x: 5, y: 5 } },
      { props: { x: 10, y: 10 }, result: { x: 10, y: 10 } },
      { props: { x: 15, y: 15 }, result: { x: 15, y: 15 } },
    ])('com x=$props.x e y=$props.y', ({ props, result }) => {
      const container = setup(props);
      const spacing = container.getByTestId('spacing');
      expect(spacing).toHaveStyle({ marginHorizontal: result.x });
      expect(spacing).toHaveStyle({ marginVertical: result.y });
      expect(container).toMatchSnapshot();
    });
  });
});
