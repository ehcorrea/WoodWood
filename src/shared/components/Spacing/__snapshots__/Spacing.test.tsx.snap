// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Spacing/> quando renderizado com x=5 e y=5 1`] = `
<View
  style={
    [
      {
        "marginHorizontal": 5,
        "marginVertical": 5,
      },
      undefined,
    ]
  }
  testID="spacing"
  x={5}
  y={5}
/>
`;

exports[`<Spacing/> quando renderizado com x=10 e y=10 1`] = `
<View
  style={
    [
      {
        "marginHorizontal": 10,
        "marginVertical": 10,
      },
      undefined,
    ]
  }
  testID="spacing"
  x={10}
  y={10}
/>
`;

exports[`<Spacing/> quando renderizado com x=15 e y=15 1`] = `
<View
  style={
    [
      {
        "marginHorizontal": 15,
        "marginVertical": 15,
      },
      undefined,
    ]
  }
  testID="spacing"
  x={15}
  y={15}
/>
`;
